### PR TITLE
Add support for serializing raw_string formats

### PR DESCRIPTION
**Pull request by mwylde**
This PR adds support for `raw_string` as an output format, which writes a single string field as UTF-8 encoded bytes.

It can be used like this:

```sql
CREATE TABLE raw_sink (
  output TEXT
) WITH (
  connector = 'kafka',
  bootstrap_servers = 'localhost:9092',
  type = 'sink',
  topic = 'outputs',
  format = 'raw_string'
);

INSERT INTO raw_sink
SELECT bid.channel
FROM nexmark;
```

When combined with UDFs utilizing serde_json functions, this allow serialization of arbitrarily complex structures that are not supported via our normal JSON support. For example, this UDF:

```rust
fn my_to_json(f: f64) -> String {
    let v = serde_json::json!({
        "my_complex": {
            "nested": f
        }
    });

    serde_json::to_string(&v).unwrap()
}
```

with this query:

```sql
INSERT INTO sink
SELECT my_to_json(avg(CAST(price as FLOAT))) from coinbase
WHERE type = 'ticker'
GROUP BY hop(interval '5' second, interval '1 minute');
```

Will produce records like 

```
{"my_complex":{"nested":25900.074265625}}
{"my_complex":{"nested":25899.815841674805}}
```

The implementation is not currently type safe, in the sense that `to_raw_string` is implemented for only valid output types (those that have a single TEXT field in their schema). We validate at the query compilation level that this is correct, but I don't see a way to have the type system enforce it.

### CODE REVIEW

File: arroyo-sql/src/types.rs [hunk 1]
```diff
@@ -241,6 +241,26 @@ impl StructDef {
...
+                    TypeDef::DataType(DataType::Utf8, _)
+                ) {
+                let field = &self.fields[0].field_ident();
+                if self.fields[0].nullable() {
+                    quote! {
+                        self.#field.as_ref().map(|v| v.as_bytes().to_vec())
+                    }
+                } else {
+                    quote! {
+                        self.#field.as_bytes().to_vec()
```
Comment (jacksonrnewhouse) [inline arroyo-sql/src/types.rs:257]: I think to_raw_string() is expecting an `Option<Vec<u8>>`

**Review by jacksonrnewhouse — CHANGES_REQUESTED** [review]
Can you add a compilation test with a field like `raw_value TEXT NOT NULL`?

File: arroyo-sql/src/types.rs (hunk 1 above)
Comment (mwylde) [reply arroyo-sql/src/types.rs]: Good catch, fixed and added a test

**Review by jacksonrnewhouse — APPROVED** [review]
